packages/core/src/common/quick-pick-service.ts

```diff
@@ -101,17 +112,62 @@ export interface QuickPickValue<V> extends QuickPickItem {
 }
 
 export interface QuickInputButton {
-    iconPath?: URI | { light: URI; dark: URI } | { id: string };
+    iconPath?: URI | { light?: URI | Uri; dark: URI | Uri } | { id: string };
     iconClass?: string;
     tooltip?: string;
+    /**
+     * Whether the button should be visible even when the user is not hovering.
+     */
+    alwaysVisible?: boolean;
+}
+
+export interface NormalizedQuickInputButton extends QuickInputButton {
+    iconPath?: { light?: Uri, dark: Uri };
+}
+
+export namespace QuickInputButton {
+    export function normalize(button: undefined): undefined;
+    export function normalize(button: QuickInputButton): NormalizedQuickInputButton;
+    export function normalize(button?: QuickInputButton): NormalizedQuickInputButton | undefined {
+        if (!button) {
+            return button;
+        }
+        let iconPath: NormalizedQuickInputButton['iconPath'] = undefined;
+        if (button.iconPath instanceof URI) {
+            iconPath = { dark: button.iconPath['codeUri'] };
+        } else if (button.iconPath && 'dark' in button.iconPath) {
+            const dark = Uri.isUri(button.iconPath.dark) ? button.iconPath.dark : button.iconPath.dark['codeUri'];
+            const light = Uri.isUri(button.iconPath.light) ? button.iconPath.light : button.iconPath.light?.['codeUri'];
+            iconPath = { dark, light };
+        }
+        return {
+            ...button,
+            iconPath,
+        };
+    }
 }
 
 export interface QuickInputButtonHandle extends QuickInputButton {
     index: number; // index of where they are in buttons array if QuickInputButton or -1 if QuickInputButtons.Back
 }
 
+export enum QuickInputHideReason {
+    /**
+     * Focus was moved away from the input, but the user may not have explicitly closed it.
+     */
+    Blur = 1,
+    /**
+     * An explicit close gesture, like striking the Escape key
+     */
+    Gesture = 2,
+    /**
+     * Any other reason
+     */
+    Other = 3,
+}
+
 export interface QuickInput {
-    readonly onDidHide: Event<void>;
+    readonly onDidHide: Event<{ reason: QuickInputHideReason }>;
```

Comment:
This reflects a change in the type emitted by the Monaco class we delegate to to implement `QuickInput`. For most of the event emitters, we have getters that return the delegate's events, so the fact that the delegate's event was no longer `void` caused a type issue, and it seemed more future-oriented to expose the new argument rather than suppress it.